ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs

```diff
@@ -22,12 +22,22 @@
 //!       * Have a float-type as the destination type operand.
 //!       * Have a destination type operand at least as big as the type of the source operand.
 //!   * [Const::Int]s cannot use more bits than the corresponding [Ty::Integer] type.
+//!   * [super::Inst] operands refer to values which have been previously defined.
 
 use super::{BinOp, BinOpInst, Const, GuardInst, Inst, Module, Operand, Ty};
 
 impl Module {
     pub(crate) fn assert_well_formed(&self) {
         for (iidx, inst) in self.iter_skipping_insts() {
+            inst.map_packed_operand_locals(self, &mut |x| {
+                let i = self.inst_all(x);
+                if i.def_type(self).is_none() {
+                    panic!(
```

Comment:
Can we add a `#[should_panic]` test for this?

Reply:
I don't think we can. I had a go but it looks like this sort of error is already caught by the parser:

`.../jit_ir/parser.rs:559:  self.error_at_span(span, &format!("Undefined local operand '%{idx}'"))`

So I think the test suite is too clever for me to build this kind of malformed IR

Reply:
Indeed, I did think I'd implemented this somewher! One could create the mall-formed IR by hand, but it's annoying to do so. I think this is OK as-is.